test/e2e/playwright/swap/pageObjects/swap-page.ts

```diff
@@ -109,18 +166,29 @@ export class SwapPage {
   }
 
   async selectTokenFromList(symbol: string) {
-    let searchItem;
+    let count;
+    // wait for the list to populate
     do {
-      searchItem = await this.tokenList.first().textContent();
-    } while (searchItem !== symbol);
-
-    await this.tokenList.first().click();
+      count = await this.tokenList.count();
+      await this.page.waitForTimeout(500);
+    } while (count !== (await this.tokenList.count()));
+
+    await this.tokenSearch.fill(symbol);
+    const regex = new RegExp(`^${symbol}$`, 'u');
+    const searchItem = await this.tokenList.filter({ hasText: regex });
+    await searchItem.click({ timeout: 5000 });
   }
 
   async waitForSearchListToPopulate(symbol: string): Promise<void> {
     let searchItem;
     do {
       searchItem = await this.tokenList.first().textContent();
+      const listCount = await this.tokenList.count();
```

Comment:
potential infinite loop

Reply:
good catch. THis was old code that was no longer needed and I removed it